### ISSUE TITLE
Results Metadata

### ISSUE THREAD

**Issue by JohnNKing** [CDCgov/trusted-intermediary#650]
# Story
As a lab, to troubleshoot issues and generate usage analytics for NBS results, I need metadata to be captured for results messages sent through ReportStream.

## Pre-conditions
- Results work completed

## Acceptance Criteria
- [x] The lab can be provided with detailed errors of their results messages.
- [x] The results metadata is available in the form of FHIR OperationOutcome
- [x] Results metadata can be accessed for at least the past 90 days.

## Tasks
- See Engineering Section

### Research
n/a

###  Engineering
- [x] Review metadata implementation as used by the orders flow (see #610, #672, #611)
- [x] Storage of metadata
  - [x] Refactor Metadata to handle Orders and Results. 
  - [x] Update Results flow to store current PartnerMetadata.
  - [x] Add Storage Typing to distinguish between Orders and Result metadata
    - [x] Update Db objects to handle Metadata updates
    - [x] Update result and order flow to extract MessageType(ORU or ORM/OML) and save it under PartnerMetadata 
- [x] Retrieval of metadata
  - [x] Test retrieving Result Metadata works with current metadata endpoint
    - [x] If not working, add support for Results Metadata.
  - [ ] Test retrieving Orders works as expected.
- [ ] Blocked by #1001

## Definition of Done
- [ ] Documentation tasks completed
  - [x] Documentation and diagrams created or updated
    - [x] Implementation guide (`/ig` folder)
    - [x] ADRs (`/adr` folder)
    - [x] Main `README.md`
    - [x] Other READMEs in the repo
    - [x] If applicable, update the `ReportStream Setup` section in `README.md`
  - [x] Threat model updated
  - [x] API documentation updated
  - [x] Source code documentation created when the code is not self-documenting
- [ ] Code quality tasks completed
  - [x] Logging added where useful
  - [x] Code refactored for clarity and no design/technical debt
  - [x] Adhere to separation of concerns; code is not tightly coupled, especially to 3rd party dependencies
  - [ ] Code is reviewed or developed by pair; 1 approval is needed but consider requiring an outside-the-pair reviewer
  - [x] Code quality checks passed
- [x] Security & Privacy tasks completed
  - [x] Security & privacy gates passed
- [x] Testing tasks completed
  - [x] Load tests passed
  - [x] Unit test coverage of our code >= 90%
- [ ] Build & Deploy tasks completed
  - [ ] Build process updated
  - [ ] API(s) are versioned
  - [ ] Feature toggles created and/or deleted.  Document the feature toggle
  - [ ] Source code is merged to the main branch

## Research Questions
- *Optional: Any initial questions for research*

## Decisions
- *Optional: Any decisions we've made while working on this story*

## Notes
- MVP-REQ-27: Intermediaries will provide metadata required, including elements to calculate timeliness metrics

**Comment by scleary1cs** [comment]
This Story helps complete the requirement:
MVP-REQ-27 - Intermediaries will provide metadata required, including elements to calculate timeliness metrics

**Comment by scleary1cs** [comment]
This Story helps complete the requirement:
MVP-REQ-29 - Each Intermediary shall have a dashboard for some higher level internal monitoring metrics – team who is troubleshooting and supporting would need access to these different views of the system

**Comment by JohnNKing** [comment]
@JohnNKing Need new story for connection pooling? Which will block this one
